### PR TITLE
task: Add CONTRIBUTING.md with development setup instructions

### PR DESCRIPTION
**Pull request by Eptis**
### Changes

Added a CONTRIBUTING.md with instructions to setup your development environment as requested in https://github.com/plausible/analytics/issues/176

### Tests
- [x] This PR does not require tests

### Changelog
- [x] This PR does not make a user-facing change

### Documentation
- [x] Added a CONTRIBUTING.md with instructions to setup your development environment.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+## Development setup
+
+The easiest way to get up and running is to [install](https://docs.docker.com/get-docker/) and use Docker for running both Postgres and Clickhouse.
+
+### Start the environment:
+
+1. Run both `make postgres` and `make clickhouse`.
+2. Run `mix ecto.create`. This will create the required databases in both Postgres and Clickhouse.
+3. Run `mix ecto.migrate` to build the database schema.
+4. `cd` into `/assets` using `cd /assets` and run `npm ci` to install the required node dependencies.
```
Comment (ukutaht) [inline CONTRIBUTING.md:12]: Simpler is to run `npm ci --prefix assets` from the project root

**Review by ukutaht — APPROVED** [review]
Everything looks good to me!

I only have further improvements to suggest but these are by no means blockers for this to be merged:
* Eventually would be nice to have a `docker-compose.yml` file for development so we could just spin up the entire app with `docker-compose up`. The tricky part there is figuring out how to attach volumes so live reloading and development works OK.
* For development it would be very useful to take a daily/weekly database backup of the live demo (https://plausible.io/plausible.io) and have a simple task to restore that data into your local database

File: CONTRIBUTING.md (hunk 1 above)
Comment (ukutaht) [reply CONTRIBUTING.md:12]: Once this is changed I'll merge

**Comment by Eptis** [comment]
@ukutaht with regards to docker compose, do you mean to actually run the phoenix application in Docker as well? I've had some issues in the past with compilation performance of Elixir in Docker in the past, and I don't believe copying over compiled files into docker is possible because of different compilation architectures. But I could be wrong here 😬

**Pull request merged**